main.js

```diff
@@ -0,0 +1,65 @@
+import {
+  validEmail,
+  validPassword,
+  validNickname,
+  validConfirmPassword,
+} from "./utils/valid.js";
+
+const authForm = document.querySelector(".auth_form");
+const authInputs = document.querySelectorAll(".auth_input");
+const password = document.getElementById("login-password");
+const confirmPassword = document.getElementById("login-confirm_password");
+const submitButton = document.querySelector(".submit-button");
+
+authForm.addEventListener("focusout", focusoutAuthInput);
+authForm.addEventListener("keyup", keyupConfirmPassword);
+authForm.addEventListener("click", setIsViewAboutPassword);
+
+submitButton.addEventListener("click", (e) => {
+  if (e.detail === 0) e.preventDefault();
+  console.log("click");
+});
+
+function setIsViewAboutPassword(e) {
+  const { target } = e;
+  const { parentElement } = target;
+
+  if (!parentElement.classList.contains("view-password")) return;
+  if (target.tagName !== "IMG") return;
+
+  const hiddenImg = parentElement.querySelector(".hidden");
+  console.dir(target);
+  console.dir(hiddenImg);
+  target.classList.toggle("hidden");
+  hiddenImg.classList.toggle("hidden");
+
+  const password = parentElement.parentElement.querySelector(".auth_input");
+  const type = password.getAttribute("type");
+  password.setAttribute("type", type === "text" ? "password" : "text");
+}
+
+function focusoutAuthInput(e) {
+  const target = e.target;
+
+  if (!target.classList.contains("auth_input")) return;
+  if (target.id === "login-id") validEmail(target);
+  else if (target.id === "login-nickname") validNickname(target);
+  else if (target.id === "login-password") validPassword(target);
+  toggleButtonState();
+}
```

Comment:
```suggestion
const validator = {
  "login-id" : validEmail,
  "login-nickname" : validNickname,
  "login-password":validPassword
}
function focusoutAuthInput(e) {
  const target = e.target;

  if (!target.classList.contains("auth_input")) return;
  const validate = validator[target.id];
  if ( validate)  validator[target.id](target);
  toggleButtonState();
}
```
단일 비교인경우 switch case 문으로 비교하는게 판단이 더 쉽지만,
객체로 맵핑하는 경우 조건이 더 추가되거나 변경될 때에도 객체만 수정하고 연결된 함수는 그대로 쓸 수 있어서 확장, 수정이 유연한 장점이 있습니다.

Reply:
객체로 함수를 저장해서 맵핑하는걸 생각 안해봤는데, 좋은 방법을 알게되었습니다. 감사합니다. 적용해보겠습니다!!